package com.integrador.rocket.roadmap.infra;

import io.swagger.v3.oas.models.Components;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Contact;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.security.SecurityScheme;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SpringDocConfiguration {

    @Bean
    public OpenAPI customOpenApi() {
        return new OpenAPI()
                .components(new Components()
//                        .addSecuritySchemes(
//                                "bearer-key", new SecurityScheme().type(SecurityScheme.Type.HTTP).scheme("bearer").bearerFormat("JWT")
//                        )
                )
                .info(new Info()
                        .title("JOURNET")
                        .description("ENDPOINTS DISPONIBLES")
                        .contact(new Contact()
                                .name("MZP")
                                .email("mz2458594@gmail.com")
                        )
                )
                ;
    }

}
